main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"net/url"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/observatorium/observatorium/internal"
+	"github.com/observatorium/observatorium/internal/proxy"
+	"go.uber.org/automaxprocs/maxprocs"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/route"
```

Comment:
Personally, I'd prefer go-chi as router here, which is a bit more maintained in terms of features, I believe. Not sure how common it is to use Prometheus' route. To me it seems more like an externally exposed internal package, but I can be wrong here.

Reply:
I don't have strong feelings about the router at this stage. We don't use any of the offered features right now, I just thought given the requirements, we would need a more feature-complete router than `http.ServeMux`. That being said, I have chosen Prometheus router because it's a thin wrapper around https://github.com/julienschmidt/httprouter , which is the one I'd prefer to use. Again, I don't have any strong feelings about it.

If you all think, we should go with `go-chi`, I can quickly make the necessary changes.

Reply:
Makes sense. The problem with httprouter is that it doesn't use the stdlib http handlers. My question mostly was about Prometheus router being used by other projects though. :)

Reply:
I think itis quite easy to change overtime - I personally don't have strong opinion other than choosing one (: Something that uses native HTTP handlers would be amazing.

Reply:
Alright, I'll check `go-chi` then, I like the idea to be able to use native HTTP handlers.